### PR TITLE
Tag app logs with "APP" source type

### PR DESCRIPTION
**Pull request by tcdowney**
## Is there a related GitHub Issue?
Issue: https://github.com/cloudfoundry/korifi/issues/1431

## What is this change about?
Tag app logs with "APP" source type

## Does this PR introduce a breaking change?
No

## Acceptance Steps
1. `cf push` an app that emits logs
2. Run `cf logs MY_APP` and see output like this

```console
$ cf logs node2
Retrieving logs for app node2 in org o / space s as cf-admin...

   2022-09-01T15:02:52.89-0600 [APP/] OUT Console output from test-node-app
   2022-09-01T15:02:52.89-0600 [APP/] OUT
   2022-09-01T15:02:53.89-0600 [APP/] OUT Console output from test-node-app
   2022-09-01T15:02:53.89-0600 [APP/] OUT
   2022-09-01T15:02:54.89-0600 [APP/] OUT Console output from test-node-app
   2022-09-01T15:02:54.89-0600 [APP/] OUT
   2022-09-01T15:02:55.89-0600 [APP/] OUT Console output from test-node-app
   2022-09-01T15:02:55.89-0600 [APP/] OUT
```

## Tag your pair, your PM, and/or team
@acosta11

### CODE REVIEW

**Comment by tcdowney** [comment]
I modified the node app that was checked in to print a log line every second to make sure the e2e tests had enough `APP` logs. Basically just added this to it:

```
function logger() {
    console.log('Console output from test-node-app');
}

setInterval(logger, 1000);
```

**Review by acosta11 — APPROVED** [review]
LGTM

File: api/repositories/pod_repository.go [hunk 1]
```diff
@@ -405,6 +399,21 @@ func (r *PodRepo) GetRuntimeLogsForApp(ctx context.Context, logger logr.Logger,
 	return appLogs, nil
 }
 
+func lineToAppLogRecord(line []byte) LogRecord {
```
Comment (tcdowney) [inline api/repositories/pod_repository.go:402]: Extracted this to appease CodeClimate 😅 

To be honest, it was probably right about this function being too long.

**Comment by tcdowney** [comment]
An unrelated test panicked:

```
Test Panicked
  In [It] at: /home/runner/go/pkg/mod/github.com/onsi/gomega@v1.20.2/internal/async_assertion.go:75

  runtime error: invalid memory address or nil pointer dereference
```

Went ahead and included a commit to fix that.

**Comment by tcdowney** [comment]
I think the e2e logging test I introduced is flaky -- going to take another look at it.

**Pull request merged**